lib/multicont_stubs.c

```diff
@@ -26,14 +38,22 @@ value multicont_promote(value k) {
 
   value null_stk = Val_ptr(NULL);
 
+#if MULTICONT52
+  r = caml_alloc_2(Cont_tag, null_stk, null_stk);
+#else
   r = caml_alloc_1(Cont_tag, null_stk);
+#endif
 
   // Move the stack from [k] to [r]
   {
     // Prevent the GC from running between [caml_continuation_use] and
     // [caml_continuation_replace]
     CAMLnoalloc;
     caml_continuation_replace(r, Ptr_val(caml_continuation_use(k)));
+#if MULTICONT52
+    Field(r, 1) = Field(k, 1); // TODO(dhil): Do we need to use an
```

Comment:
I think these uses of `Field` (rather than `Store_field`) are valid, because the GC cannot run between the small alloc and the direct assignment, but given that it's an FFI violation of rule 6, it's probably worth an explicit comment?

Reply:
Thanks David! I appreciate the help.